### PR TITLE
Navigation

### PR DESCRIPTION
**Pull request by illegalDisease**
A new util to handle navigation applications properly

### CODE REVIEW

File: lib/navigation.js [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+ * @author Ozcan Ovunc <ozcan.ovunc@smartface.io>
+ * @copyright Smartface 2018
+ */
+const System = require('sf-core/device/system');
+const Application = require("sf-core/application");
+const Menu = require("sf-core/ui/menu");
+const MenuItem = require("sf-core/ui/menuitem");
+const langChecker = require("./base/langchecker")("navigation");
+
+langChecker.check(["cancel", "chooseNavigationApp"]);
```
Comment (alperozisik) [inline lib/navigation.js]: Add them in empty project template

File: lib/navigation.js [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+
+var showNavigationMenuForIOS = (function() {
```
Comment (alperozisik) [inline lib/navigation.js]: use `const` here

File: lib/navigation.js [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+ *  });
+ * 
+ */
+exports.showNavigationMenu = function({ page, transportType = "d", location }) {
+    if (System.OS === "iOS") {
+        showNavigationMenuForIOS({ page, transportType, location });
+    }
+    else {
+
+        // TODO: Handle transportType
```
Comment (alperozisik) [inline lib/navigation.js]: Still, **TODO** to do?

File: lib/navigation.js [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+        Application.call(
+            `geo:${location.latitude},${location.longitude}?q=${location.latitude},${location.longitude}`,
+            null, // data
+            _ => _, // onSuccess
```
Comment (alperozisik) [inline lib/navigation.js]: `_`

File: lib/navigation.js [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+            _ => _, // onFailure
```
Comment (alperozisik) [inline lib/navigation.js]: `_`

File: lib/navigation.js [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+ * @function
+ * @param {Object} page - The main object of current page.
+ * @param {string} transportType - Your way of travel, driving or walking. Accepted paramters = "d", "w"
+ * @param {Object} location - Destination location which contains latitude and longitude
+ * @example
+ * const navigation = require("sf-extension-utils/lib/navigation");
+ * navigation.showNavigationMenu({
+ *      page,
+ *      transportType: "d",
+ *      location: {
```
Comment (alperozisik) [inline lib/navigation.js]: Why Moskow?

Use the same location that we are using in the maps examples.

**Comment by alperozisik** [comment]
1) Apple maps can be uninstalled. The library should check its availability.
img_61a6b8ab0fe6-1 
2) If there is a single choice of the navigation app, it should directly open it without asking
3) The call of the function should return a promise
  a) If the navigation app is opened, should resolve with the choice (if possible)
  b) If the user closes the menu, it should reject with `"user canceled"`
  c) If there are no apps to navigate, then it should reject with `"no navigation choices"`

File: lib/navigation.js (hunk 1 above)
Comment (illegalDisease) [reply lib/navigation.js]: Opened [Pull Request](https://github.com/smartface/smfc-empty-workspace/pull/27)

File: lib/navigation.js (hunk 6 above)
Comment (illegalDisease) [reply lib/navigation.js]: Changed it to default location which used on MapView docs, which shows Arthur Ross Pinetum Playground.

File: lib/navigation.js (hunk 3 above)
Comment (illegalDisease) [reply lib/navigation.js]: done.

**Pull request merged**